hivemq-edge/src/frontend/src/extensions/datahub/designer/behavior_policy/BehaviorPolicyNode.utils.ts

```diff
@@ -0,0 +1,43 @@
+import { Node } from 'reactflow'
+
+import { BehaviorPolicyData, DryRunResults } from '@datahub/types.ts'
+import { PolicyCheckErrors } from '@datahub/designer/validation.errors.ts'
+import { BehaviorPolicy, BehaviorPolicyBehavior, BehaviorPolicyOnTransition } from '@/api/__generated__'
+
+export function checkValidityModel(behaviorPolicy: Node<BehaviorPolicyData>): DryRunResults<BehaviorPolicyBehavior> {
+  if (!behaviorPolicy.data.model) {
+    return {
+      node: behaviorPolicy,
+      error: PolicyCheckErrors.notConfigured(behaviorPolicy, 'model'),
+    }
+  }
+  return {
+    node: behaviorPolicy,
+    data: {
+      arguments: behaviorPolicy.data.arguments,
+      id: behaviorPolicy.data.model,
+    },
+  }
+}
+
+export function checkValidityBehaviorPolicy(
+  node: Node<BehaviorPolicyData>,
+  client: DryRunResults<string, never>,
+  model: DryRunResults<BehaviorPolicyBehavior, never>,
+  transitions: DryRunResults<BehaviorPolicyOnTransition, never>[]
+): DryRunResults<BehaviorPolicy, unknown> {
+  return {
+    node: node,
+    data: {
+      behavior: model.data as BehaviorPolicyBehavior,
+      // TODO[19466] Id should be user-facing; Need to fix before merging!
+      id: node.id,
```

Comment:
issue(blocking): FYI found a TODO that might need to be resolved before this gets merged

Reply:
A few of them. They will be fixed in the mentioned ticket number (19466)